dex/networks/eth/txdata.go

```diff
@@ -155,26 +126,26 @@ func (t *txDataHandlerV0) parseInitiateData(calldata []byte) ([]*Initiation, err
 		_ = swapv0.ETHSwapInitiation(initiations[0])
 	}
 
-	toReturn := make([]*Initiation, 0, len(initiations))
+	toReturn := make(map[[SecretHashSize]byte]*Initiation)
```

Comment:
Should still allocate `len(initiations)`, but it's not gonna hold up my approval.